torchani/data/new.py

```diff
@@ -0,0 +1,458 @@
+import numpy as np
+import torch
+import functools
+from ._pyanitools import anidataloader
+import importlib
+
+PKBAR_INSTALLED = importlib.util.find_spec('pkbar') is not None
+if PKBAR_INSTALLED:
+    import pkbar
+
+
+def find_threshold(file_path, batch_size, threshold_max=100):
+    """Find resonable threshold to split chunks
+    Arguments:
+        file_path (str): Path to one hdf5 files.
+        batch_size (int): batch size.
+        threshold_max (int): max threshould to test.
+    """
+    ds = CachedDataset(file_path=file_path, batch_size=batch_size)
+    ds.find_threshold(threshold_max + 1)
+
+
+class CachedDataset(torch.utils.data.Dataset):
+    """ Cached Dataset which is shuffled once, but the dataset keeps the same at every epoch
+    Arguments:
+        file_path (str): Path to one hdf5 files.
+        batch_size (int): batch size.
+        chunk_threshold (int): threshould to split batch into chunks.
+        species_order (str): a string which specify how species are transfomed to int.
+            for example: 'HCNO' means {'H': 0, 'C': 1, 'N': 2, 'O': 3}
+        subtract_self_energies (bool): whether subtract self energies from energies
+        self_energies (list): if `subtract_self_energies` is True, the order should keep
+            the same as `species_order`.
+            for example :[-0.600953, -38.08316, -54.707756, -75.194466] will be converted
+            to {'H': -0.600953, 'C': -38.08316, 'N': -54.707756, 'O': -75.194466}
+
+    Returns:
```

Comment:
No, a class does not return anything. You need to say: "the resulting dataset will be...."